src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceSetWrapperBase.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.OData;
+
+namespace Microsoft.AspNet.OData.Formatter.Deserialization
+{
+    /// <summary>
+    /// Encapsulates an <see cref="ODataResourceSet"/> and the <see cref="ODataResource"/>'s that are part of it.
+    /// </summary>
+    public abstract class ODataResourceSetWrapperBase : ODataItemBase
+    {
+        /// <summary>
+        /// /
+        /// </summary>
+        public ResourceSetType ResourceSetType;
```

Comment:
>public ResourceSetType ResourceSetType; [](start = 8, length = 39)

Don't expose a public filed. use property.

It seems you can make it abstract as below;
internal abstract ResourceSetType ResourceSetType {get;}